resolver/src/dns_lru.rs

```diff
@@ -35,11 +35,47 @@ impl LruValue {
 }
 
 #[derive(Debug)]
-pub(crate) struct DnsLru(LruCache<Query, LruValue>);
+pub(crate) struct DnsLru {
+    cache: LruCache<Query, LruValue>,
+    /// An optional minimum TTL value for positive responses.
+    ///
+    /// Positive responses with TTLs under `min_positive_ttl` will use
+    /// `` instead.
+    min_positive_ttl: Duration,
+    /// An optional minimum TTL value for negative (`NXDOMAIN`) responses.
+    ///
+    /// `NXDOMAIN` responses with TTLs under `min_negative_ttl` will use
+    /// `min_negative_ttl` instead.
+    min_negative_ttl: Duration,
+}
+
+#[derive(Copy, Clone, Debug, Default)]
+pub(crate) struct TtlConfig {
+    /// An optional minimum TTL value for positive responses.
+    ///
+    /// Positive responses with TTLs under `min_positive_ttl` will use
+    /// `` instead.
+    pub min_positive_ttl: Option<Duration>,
+    /// An optional minimum TTL value for negative (`NXDOMAIN`) responses.
+    ///
+    /// `NXDOMAIN` responses with TTLs under `min_negative_ttl` will use
+    /// `min_negative_ttl` instead.
+    pub min_negative_ttl: Option<Duration>,
+}
 
 impl DnsLru {
     pub(crate) fn new(capacity: usize) -> Self {
-        DnsLru(LruCache::new(capacity))
+        Self::with_ttls(capacity, TtlConfig::default())
+    }
```

Comment:
IMO it would be better to require users of `DnsLru` to do this manually, rather than defaulting here. Otherwise it would be too easy to write code that doesn't honor the TTL configuration. In other words, IMO this `new()` should just be removed and `with_ttls` should just be renamed `new()`.

Otherwise, LGTM.

Reply:
Done in 9de5612cc74d53e80635d23ee808b45586f4f420 :)